ops/terraform/services/server/main.tf

```diff
@@ -0,0 +1,219 @@
+    nonsensitive(data.aws_ssm_parameters_by_path.nonsensitive_service.values)
+  )
+  nonsensitive_service_config = {
+    for key, value in local.nonsensitive_service_map
+    : split("/", key)[5] => value
+  }
+
+  enterprise_tools_security_group = local.nonsensitive_common_config["enterprise_tools_security_group"]
+  management_security_group       = local.nonsensitive_common_config["management_security_group"]
+  vpn_security_group              = local.nonsensitive_common_config["vpn_security_group"]
+  kms_key_alias                   = local.nonsensitive_common_config["kms_key_alias"]
+  ssh_key_pair                    = local.nonsensitive_common_config["key_pair"]
+  vpc_name                        = local.nonsensitive_common_config["vpc_name"]
+
+  lb_is_public                   = local.nonsensitive_service_config["lb_is_public"]
+  lb_ingress_port                = local.nonsensitive_service_config["lb_ingress_port"]
+  lb_egress_port                 = local.nonsensitive_service_config["lb_egress_port"]
+  lb_vpc_peerings                = jsondecode(local.nonsensitive_service_config["lb_vpc_peerings_json"])
+  asg_min_instance_count         = local.nonsensitive_service_config["asg_min_instance_count"]
+  asg_max_instance_count         = local.nonsensitive_service_config["asg_max_instance_count"]
+  asg_max_warm_instance_count    = local.nonsensitive_service_config["asg_max_warm_instance_count"]
+  asg_desired_instance_count     = local.nonsensitive_service_config["asg_desired_instance_count"]
+  asg_instance_warmup_time       = local.nonsensitive_service_config["asg_instance_warmup_time"]
+  launch_template_instance_type  = local.nonsensitive_service_config["launch_template_instance_type"]
+  launch_template_volume_size_gb = local.nonsensitive_service_config["launch_template_volume_size_gb"]
+
+  # ephemeral environment determination is based on the existence of the ephemeral_environment_seed
+  # in the common hierarchy
+  seed_env         = lookup(local.nonsensitive_common_config, "ephemeral_environment_seed", null)
+  is_ephemeral_env = local.seed_env == null ? false : true
+
+  env_config = {
+    default_tags = local.default_tags,
+    vpc_id       = data.aws_vpc.main.id,
+    azs          = local.azs
+  }
+  cw_period       = 60 # Seconds
+  cw_eval_periods = 3
+
+  ami_id = data.aws_ami.main.image_id
+
+  create_server_lb_alarms   = !local.is_ephemeral_env || var.force_create_server_lb_alarms
+  create_server_metrics     = !local.is_ephemeral_env || var.force_create_server_metrics
+  create_server_slo_alarms  = (local.create_server_metrics && !local.is_ephemeral_env) || var.force_create_server_slo_alarms
+  create_server_log_alarms  = !local.is_ephemeral_env || var.force_create_server_log_alarms
+  create_server_dashboards  = (local.create_server_metrics && !local.is_ephemeral_env) || var.force_create_server_dashboards
+  create_server_disk_alarms = !local.is_ephemeral_env || var.force_create_server_disk_alarms
```

Comment:
A note for later changeset(s) (potentially as part of forthcoming BFD-2554), this override functionality might still be useful and remain, but the determination for creating these resources ought to come from the environment definition as set in `base`, i.e. SSM parameter store lookups. No change requested or required 😉.